src/mcp/server/auth/provider.py

```diff
@@ -174,6 +174,27 @@ async def authorize(
         """
         ...
 
+    async def has_client_consent(self, client: OAuthClientInformationFull) -> bool:
```

Comment:
todo: if client_consent_required, these need to be required, but otherwise they should be optional.  open to suggestions here. i think a default implementation of `return True` means we'd solve for being optional, but not for required.